### PR TITLE
Clean up eureka-client deprecations

### PR DESCRIPTION
**Pull request by ZIRAKrezovic**
Fixes: #4133

### CODE REVIEW

File: spring-cloud-netflix-eureka-client/src/main/java/org/springframework/cloud/netflix/eureka/http/RestTemplateTransportClientFactory.java [hunk 1]
```diff
@@ -209,11 +209,6 @@ public HttpStatusCode getStatusCode() throws IOException {
 			return response.getStatusCode();
 		}
 
-		@Override
```
Comment (OlgaMaciaszek) [inline spring-cloud-netflix-eureka-client/src/main/java/org/springframework/cloud/netflix/eureka/http/RestTemplateTransportClientFactory.java:212]: We cannot remove public methods in a non-major release. Please revert this change.

**Review by OlgaMaciaszek — CHANGES_REQUESTED** [review]
Hello @ZIRAKrezovic. Thanks for the PR. Have added a comment. Please address.

File: spring-cloud-netflix-eureka-client/src/main/java/org/springframework/cloud/netflix/eureka/http/RestTemplateTransportClientFactory.java (hunk 1 above)
Comment (ZIRAKrezovic) [reply spring-cloud-netflix-eureka-client/src/main/java/org/springframework/cloud/netflix/eureka/http/RestTemplateTransportClientFactory.java:212]: There is a default implementation on the interface level that delegates to getStatusCode(), so technically nothing is removed.

Comment (ZIRAKrezovic) [reply spring-cloud-netflix-eureka-client/src/main/java/org/springframework/cloud/netflix/eureka/http/RestTemplateTransportClientFactory.java]: But in any case, I have reverted this change.

**Comment by ZIRAKrezovic** [comment]
Hi @OlgaMaciaszek. Thanks for the review, I believe I have addressed your feedback.

File: spring-cloud-netflix-eureka-client/src/main/java/org/springframework/cloud/netflix/eureka/http/RestTemplateTransportClientFactory.java (hunk 1 above)
Comment (OlgaMaciaszek) [reply spring-cloud-netflix-eureka-client/src/main/java/org/springframework/cloud/netflix/eureka/http/RestTemplateTransportClientFactory.java]: You're right. Have missed that the implementation is same anyhow. You can remove it.

Comment (ZIRAKrezovic) [reply spring-cloud-netflix-eureka-client/src/main/java/org/springframework/cloud/netflix/eureka/http/RestTemplateTransportClientFactory.java:212]: I have re-removed it.

**Review by OlgaMaciaszek — APPROVED** [review]
LGTM.
